test/file-base.js

```diff
@@ -119,3 +119,27 @@ describe('hashSha256', () => {
     )
   })
 })
+
+describe('cached Hash', () => {
+  it('hash is cache to file', async () => {
+    const path_ = 'test/fixtures/sample-cyrillic-encoding.csv'
+    const file = data.open(path_)
+
+    let hash = await file.hash()
+    assert.strictEqual(
+      file._computedHashes['md5'],
+      '37d3a5159433f0977afb03d01d4bde6e'
+    )
+
+    let cachedhash = await file.hash()
+    assert.strictEqual(cachedhash, '37d3a5159433f0977afb03d01d4bde6e')
+
+    let hash256 = await file.hash('sha256')
+    assert.strictEqual(
+      file._computedHashes['sha256'],
+      '8eff5a7815864615309d48035b461b79aa1bdc4402924e97fc66e123725214fd'
+    )
+    assert.strictEqual(hash256, file._computedHashes['sha256'])
+    assert.strictEqual(hash, file._computedHashes['md5'])
```

Comment:
How we can convinced that the hashes are actually cached not regenerated? I mean if don't change the file content it will always return correct hash, this what is asserted in this test, but how you can make sure that it is cached?

Reply:
This is quite tricky to automate in the tests unless I add a console statement. It works via manual testing though

Reply:
Testing for things that don't change behavior is dubious and typically wrong. Suggest you just make sure to test that you get the same hash twice for the same object when calling `hash()`, this way you ensure the cache doesn't *break* anything.

Reply:
@shevron Since we are already testing hash function before, it becomes quite redundant. Suggest removing this unit test entirely

Reply:
You should consider what can cache break. Adding another similar test that just calls the same function twice (`file.hash('sha256')`) is super easy. Doing it has the potential of catching a type of problem that is very possible once cache is implemented - that the cached version is not valid but somehow still returned. Cheap to test, potentially high value. I think you should add this kind of test.

Reply:
@shevron alright, I'll add it. Thanks for the suggestion